### ISSUE TITLE
"Additivity check failed in TreeExplainer" Fixed by Running in Batches

### ISSUE THREAD

**Issue by RyanWendt** [shap/shap#2482]
I was getting the above error when using the TreeExplainer and the default "tree_path_dependent" option with a Scikit-Learn Gradient Boosting binary classification model. I have found a work around to the problem for my code obtaining the shap values from the feature set in more manageable 1 million at a time batches. I'm not sure what's happening on the back end of the shap_values function, but it looks like the shap values fail to calculate properly when the dataset fed into it in one function call is too large.  I wanted to post this writeup to help others and also to hopefully bring it to the author's attention to potentially patch.

Some of the other issue threads I've seen on this topic recommend switching to the "interventional" option to fix. However, this was not a practical solution for me largely due to performance constraints among others. Moreover, when turning off the additivity check, it was clear that the additivity numbers really were way off the model's prediction (I calculated this by summing the shap values by row, adding the expected value and comparing this against the model's predicted probability with a logit transformation to get the log odds value). After implementing the above work around I performed the same check again, and the numbers matched (after rounding by a few decimal places).
